packages/monitor-v2/src/monitor-og/oSnapAutomation.ts

```diff
@@ -391,3 +523,24 @@ export const proposeTransactions = async (logger: typeof Logger, params: Monitor
   // Submit proposals.
   await submitProposals(logger, verifiedProposals, supportedModules, params);
 };
+
+export const disputeProposals = async (logger: typeof Logger, params: MonitoringParams): Promise<void> => {
+  // Get all undiscarded on-chain proposals for all monitored modules.
+  const onChainProposals = await getUndiscardedProposals(params.ogAddresses, params);
+
+  // Filter out all proposals that have been executed on-chain.
+  const unexecutedProposals = await filterUnexecutedProposals(onChainProposals, params);
+
+  // Filter out all proposals that have not passed their challenge period.
+  const lastTimestamp = await getBlockTimestamp(params.provider, params.blockRange.end);
+  const liveProposals = unexecutedProposals.filter((proposal) => !hasChallengePeriodEnded(proposal, lastTimestamp));
+
+  // Filter only supported proposals and get their parameters.
+  const supportedProposals = await getSupportedProposals(liveProposals, params);
```

Comment:
Module parameters might have changed since proposal, thus, we cannot filter supported modules the same as for proposal bot. Instead, we need to evaluate dispute support based on bond/rules parameters at the time of particular proposal.